src/less/_flag.less

```diff
@@ -39,171 +39,122 @@
-    border-right-color: transparent;
+  &.warning {
+    .flag-variant(@warning);
   }
-  &:before {
-    border-color: darken(@danger, 5%);
-    border-left-color: transparent;
+  &.danger {
+    .flag-variant(@danger);
   }
 }
 
-//== Position
 //
+// == Position ==
 //
 
-// Flag-Right
-.flag-right {
-  position: absolute;
-  top:   1em;
-  right: 0;
-  margin-right: -.5em;
-  -webkit-border-radius:0 2px 2px 0;
-     -moz-border-radius:0 2px 2px 0;
-      -ms-border-radius:0 2px 2px 0;
-       -o-border-radius:0 2px 2px 0;
-          border-radius:0 2px 2px 0;
-  &:after {
-    content: "";
-    display: block;
-    width:  0;
-    height: 0;
+.component-flag {
+
+  // Right
+  &.right {
     position: absolute;
-    z-index: -1;
-    border-width: .25em;
-    top:   1.5em;
+    top:   1em;
     right: 0;
-    margin-top:  -1px;
-    margin-right: 0;
-    border-right-color: transparent;
-    border-bottom-color: transparent;
+    margin-right: -.5em;
+    border-radius:0 2px 2px 0;
+    &:after {
+      content: "";
+      display: block;
+      width:  0;
+      height: 0;
+      position: absolute;
+      z-index: -1;
+      border-width: .25em;
+      top:   1.5em;
+      right: 0;
+      margin-top:  -1px;
+      margin-right: 0;
+      border-right-color: transparent;
+      border-bottom-color: transparent;
+    }
+    &:before {
+      top:0;
+      z-index:0;
+      border-color: inherit;
+      border-left-color: transparent;
+    }
   }
-  &:before {
-    top:0;
-    z-index:0;
-    border-color: inherit;
-    border-left-color: transparent;
-  }
-}
 
-// Flag-Left
-.flag-left {
-  position: absolute;
-  top:  1em;
-  left: 0;
-  margin-left: -.5em;
-  -webkit-border-radius:2px 0 0 2px;
-     -moz-border-radius:2px 0 0 2px;
-      -ms-border-radius:2px 0 0 2px;
-       -o-border-radius:2px 0 0 2px;
-          border-radius:2px 0 0 2px;
-  &:before {
-    content: "";
-    display: block;
-    width:  0;
-    height: 0;
+  // Left
+  &.left {
     position: absolute;
-    z-index: -1;
-    border-width: .25em;
-    top:   1.5em;
-    left:  0;
-    margin-top:  -1px;
-    margin-left: 0;
-    border-left-color: transparent;
-    border-bottom-color: transparent;
-  }
-  &:after {
-    top:0;
-    z-index:0;
-    border-color: inherit;
-    border-right-color: transparent;
+    top:  1em;
+    left: 0;
+    margin-left: -.5em;
+    border-radius:2px 0 0 2px;
+    &:before {
+      content: "";
+      display: block;
+      width:  0;
+      height: 0;
+      position: absolute;
+      z-index: -1;
+      border-width: .25em;
+      top:   1.5em;
+      left:  0;
+      margin-top:  -1px;
+      margin-left: 0;
+      border-left-color: transparent;
+      border-bottom-color: transparent;
+    }
+    &:after {
+      top:0;
+      z-index:0;
+      border-color: inherit;
+      border-right-color: transparent;
+    }
   }
-}
 
-// Flag-Top
-.flag-top {
-  top:    1em;
-  bottom: auto;
-}
+  // Top
+  &.top {
```

Comment:
Do we need this? .right and .left have `top:  1em;` by default. If you added `bottom: auto` you could do away with the need for this class.